source/slang/slang-ir-explicit-global-context.cpp

```diff
@@ -12,8 +12,108 @@ namespace Slang
 // thread-group, and wrap them up in an explicit "context"
 // type that gets passed between functions.
 
+enum class HoistableTypes : UInt
+{
+    GlobalVariable = 0b1,
```

Comment:
can we use hex instead of binary, this seems not our coding convention?

Reply:
We can use hex, although I think this may be confusing since this code is meant to function as a Enum to a bit-field (every bit is a unique element).

I could also use a bitfield to represent this data instead of indirectly through an enum?

Note: [We seem to not have coding conventions about this](docs\design\coding-conventions.md)

Reply:
I found the binary expression not very readable.
It is one of things that C/C++ supports but nobody uses.

Reply:
> I found the binary expression not very readable.
> It is one of things that C/C++ supports but nobody uses.

To clarify (to ensure I understand correctly) is the suggestion to use: binary variable initializers, hex initializers, or bit fields?

Reply:
I think it should be hex

Reply:
Just to add more information about this.
I think the reason why the binary representation is not popular is that it quickly becomes unreadable when there are a few zeros.
A common way to define bit flags is like (0x01 << 0), (0x01 << 1), and so on.